### PR TITLE
add 1.20.5 support

### PR DESCRIPTION
**Pull request by iGabyTM**
### NOT TESTED

- changes how the MC version is detected
- uses a different ComponentSerializer on 1.20.5+ for the item and lore of the item

Paper post https://forums.papermc.io/threads/important-dev-psa-future-removal-of-cb-package-relocation.1106/

### CODE REVIEW

**Review by Kqliber — APPROVED** [review]

**Comment by Euphillya** [comment]
Why revert Folia ?!

File: core/src/main/java/dev/triumphteam/gui/guis/BaseGui.java [hunk 1]
```diff
@@ -439,15 +438,6 @@ public void close(@NotNull final HumanEntity player) {
...
     public void close(@NotNull final HumanEntity player, final boolean runCloseAction) {
-        if (VersionHelper.IS_FOLIA) {
-            player.getScheduler().runDelayed(plugin, task -> {
-                this.runCloseAction = runCloseAction;
-                player.closeInventory();
-                this.runCloseAction = true;
-            }, null, 2L);
-            return;
-        }
-
```
Comment (Euphillya) [inline core/src/main/java/dev/triumphteam/gui/guis/BaseGui.java:450]: Can be replaced by a reflection:

```java
try {
    Class<?> playerClass = player.getClass();
    Method getSchedulerMethod = playerClass.getMethod("getScheduler");
    Object scheduler = getSchedulerMethod.invoke(player);
    
    Class<?> schedulerClass = scheduler.getClass();
    
    Method executeMethod = schedulerClass.getMethod("execute", Object.class, Runnable.class, Object.class, long.class);
    
    Runnable task = () -> {
        this.runCloseAction = runCloseAction;
        player.closeInventory();
        this.runCloseAction = true;
    };
    executeMethod.invoke(scheduler, plugin, task, null, 2L);
    
} catch (Exception e) {
    e.printStackTrace();
}
```

**Review by Euphillya — CHANGES_REQUESTED** [review]

**Comment by LichtHund** [comment]
> Why revert Folia ?!
Because it requires the project to be higher Java version than intended, it needs to stay on Java 8.
I was intending on doing it in a different way, the suggestion you gave is great. Ty!

**Comment by Euphillya** [comment]
> > Why revert Folia ?!
> 
> Because it requires the project to be higher Java version than intended, it needs to stay on Java 8. I was intending on doing it in a different way, the suggestion you gave is great. Ty!

Alright I understand. So reflection is the best solution.

**Review by Azoraqua — APPROVED** [review]

**Comment by Azoraqua** [comment]
> > Why revert Folia ?!
> 
> 
> 
> Because it requires the project to be higher Java version than intended, it needs to stay on Java 8.
> 
> I was intending on doing it in a different way, the suggestion you gave is great. Ty!

May I ask what the reason as to why it must stay on Java 8 despite most platforms, including Minecraft itself running on Java 17 by now?

**Comment by LichtHund** [comment]
> May I ask what the reason as to why it must stay on Java 8 despite most platforms, including Minecraft itself running on Java 17 by now?

Because the library supports all the down to Spigot 1.8.8, there are a lot of legacy plugins depending on it and a sudden Java version change would not go well with them. This will change with v4 which will fully drop legacy support.

**Pull request merged**